pkg/controller/keycloakclient/keycloakclient_reconciler.go

```diff
@@ -47,6 +47,12 @@ func (i *KeycloakClientReconciler) Reconcile(state *common.ClientState, cr *kc.K
 		desired.AddAction(i.getUpdatedClientSecretState(state, cr))
 	}
 
+	if state.DeprecatedClientSecret != nil {
+		// Delete client secret created using the previous naming scheme, i.e., keycloak-client-secret-<CLIENT_ID>.
+		// See GH issue #473 and KEYCLOAK-18346.
+		desired.AddAction(i.getDeletedDeprecatedClientSecretState(state, cr))
```

Comment:
Before adding this action we should be sure that the new ClientSecret have been already created, correct?
Probably something like `state.ClientSecret != nil`.

Reply:
Great point @andreaTP, but I think adding this will actually prevent old secret cleanup from occurring. As of my current understanding, which could be wrong:
1. The order these actions are queued implicitly addresses your scenario. If the "create new secret" action fails then none of the queued actions following it are executed, including the queued "delete old secret" action. Stated differently, the "delete old secret" action cannot occur in a single reconciliation loop unless it was [preceded](https://github.com/keycloak/keycloak-operator/blob/main/pkg/controller/keycloakclient/keycloakclient_controller.go#L160-L163) by a successful "create new secret" or "update new secret" action.
2. I believe adding this check will actually prevent the "delete old secret" from ever being queued. Specifically, during the reconciliation loop in which "create new secret" is performed, i.e. `state.ClientSecret == nil`, then unless the actual Kubernetes secret is updated, the reconciliation loop won't be triggered [again](https://github.com/keycloak/keycloak-operator/blob/main/pkg/controller/keycloakclient/keycloakclient_controller.go#L167) and subsequently, the "delete old secret" won't ever be scheduled.

Let me know your thoughts!

Reply:
@andreaTP let me know your thoughts ☝️

Reply:
Awesome, @andreaTP Is there anything else I need to do to get this merged? Thanks!

Reply:
@trevorloranger for me is fine, you now need a review from @vmuzikar